tree_sitter/binding.c

```diff
@@ -30,14 +30,42 @@ typedef struct {
   PyObject *tree;
 } TreeCursor;
 
+typedef struct {
+  PyObject_HEAD
+  uint32_t capture1_value_id;
+  uint32_t capture2_value_id;
+  int is_positive;
+} CaptureEqCapture;
+
+typedef struct {
+  PyObject_HEAD
+  uint32_t capture_value_id;
+  PyObject *string_value;
+  int is_positive;
+} CaptureEqString;
+
+typedef struct {
+  PyObject_HEAD
+  uint32_t capture_value_id;
+  PyObject *regex;
+  int is_positive;
+} CaptureMatchString;
+
 typedef struct {
   PyObject_HEAD
   TSQuery *query;
   PyObject *capture_names;
+  PyObject *text_predicates;
 } Query;
 
+typedef struct {
+  PyObject_HEAD
+  TSQueryCapture capture;
+} QueryCapture;
+
 static TSTreeCursor default_cursor = {0};
 static TSQueryCursor *query_cursor = NULL;
+static PyObject *re_module = NULL;
```

Comment:
Are you able to you make this `re_module -> re_compile`, and initialize it (+ pass through errors with a NULL return) at the top of `PyInit_binding`? That way we're not doing an import / getattr in the middle of the object allocator.

Reply:
Yep, done. Let me know if that's what you had in mind!

Reply:
Yep!